fix: only filter displayed recent models from provider sections

joachim-isaksson opened the pull request:
Problem: Models are not displayed anywhere in the model dialog if they are in the recent list but beyond position 5.

Changed the filter to use recentList (displayed recent models) instead of recents (all recent models) to fix this issue.

rekram1-node commented:
/review

rekram1-node commented:
this seems to introduce a regression in recents list in my testing, any model I try just replaces top model in recents list instead of the previous behavior

joachim-isaksson commented:
> this seems to introduce a regression in recents list in my testing, any model I try just replaces top model in recents list instead of the previous behavior

That bug is already there and fully repeatable in the dev branch even without my patch and (as far as I understand the code) has to do with setModelStore being called with active proxy objects that are still active/changing after being inserted in the store. Deep copying the objects into the store instead seems to fix it (although I can't explain exactly why, yet)

Would you prefer I fix the modelStore bug first before adding this patch again? This patch doesn't change the behavior, just made you test it a bit extra I presume :) Otherwise I can drop in my exact test case that repeats the ModelStore problem both with or without my patch.

Commit:
fix: prevent model list corruption from SolidJS reactivity

- Deep copy objects when inserting into recent/favorite stores to break reactive proxy references
- Only filter models from provider sections if actually displayed in Recent section

joachim-isaksson commented:
Will create a new pull request, apparently a force push in my own repo broke the pull request. Sorry 'bout that.
